src/NUnitFramework/tests/Attributes/TestCaseSourceTests.cs

```diff
@@ -290,6 +290,21 @@ public void HandlesExceptionInTestCaseSource_GuiDisplay(string lhs, string rhs)
             Assert.AreEqual(lhs, rhs);
         }
 
+
+        private static IEnumerable<TestCaseData> ZeroTestCasesSource() => Enumerable.Empty<TestCaseData>();
+
+        [TestCaseSource(nameof(ZeroTestCasesSource))]
```

Comment:
I know in other projects we said it was OK to use C# 6.0 features, but I don't remember if it was acceptable for the framework itself. Namely you might need to change from nameof to a hardcoded string.